src/app/content.component.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  Component,
+  Input,
+  OnInit,
+} from '@angular/core';
+import {
+  trigger,
+  state,
+  style,
+  animate,
+  transition,
+} from '@angular/animations';
+import { TranslatePipe } from '@ngx-translate/core';
+
+import { CalculateService } from './calculate.service';
+import { CONSTANTS } from './calculate.const';
+
+@Component({
+  selector: 'app-content',
+  templateUrl: './content.component.html',
+  styleUrls: ['./content.component.scss'],
+  providers: [CalculateService],
+  animations: [
+    trigger('slideDown', [
+      state('up', style({height: 0})),
+      transition(':enter', animate('.5s ease-in')),
+      transition(':leave', animate('.5s ease-out'))
+    ])
+  ]
+})
+export class ContentComponent implements OnInit {
+  
+  private _calculateService: CalculateService;
+
+  public selectedTab = 0;
+  public years: number[];
+  public ruling: boolean;
+  public options: any;
+  public output: any;
+  public choices: string[];
+  public show: boolean = false;
+  public getKeys = Object.keys;
+
+  constructor(_calculateService: CalculateService) {
+    this._calculateService = _calculateService;
+  }
+
+  ngOnInit() {
+    this.output = CONSTANTS.default.output;
+    this.options = CONSTANTS.default.options;
+    this.options.type = CONSTANTS.default.options.type;
+    this.years = CONSTANTS.years;
+    this.options.year = this.years[0];
+    this.ruling = this.options.ruling ? true : false;
+    this.options.truncate = true;
+    let choices = [];
+    Object.keys(CONSTANTS.default.input).forEach(key => {
+      choices.push(key);
+    });
+    this.choices = choices;
+    this.changeOption(true, this.options.type);
+    this._calculateService.calculateResult.subscribe(results => {
+      this.output = results;
+    });
+  }
+
+  public rename(str: string): string {
```

Comment:
this one is actually useful so we don't have to manually put on template each translation name in the format "TRANSLATION_TITLE"